Frameworks/MQTTnet.NetStandard/Server/MqttTopicFilterComparer.cs

```diff
@@ -6,39 +6,115 @@ public static class MqttTopicFilterComparer
     {
         private static readonly char[] TopicLevelSeparator = { '/' };
 
-        public static bool IsMatch(string topic, string filter)
+        public static bool IsMatch(string topic, string sub)
         {
-            if (topic == null) throw new ArgumentNullException(nameof(topic));
-            if (filter == null) throw new ArgumentNullException(nameof(filter));
+            if (string.IsNullOrEmpty(topic)) throw new ArgumentNullException(nameof(topic));
+            if (string.IsNullOrEmpty(sub)) throw new ArgumentNullException(nameof(sub));
 
-            if (string.Equals(topic, filter, StringComparison.Ordinal))
-            {
-                return true;
-            }
 
-            var fragmentsTopic = topic.Split(TopicLevelSeparator, StringSplitOptions.None);
-            var fragmentsFilter = filter.Split(TopicLevelSeparator, StringSplitOptions.None);
+            var spos = 0;
+            var slen = sub.Length;
+            var tpos = 0;
+            var tlen = topic.Length;
+                        
+            var multilevel_wildcard = false;
 
-            for (var i = 0; i < fragmentsFilter.Length; i++)
+            while (spos < slen && tpos <= tlen)
```

Comment:
`tpos' must be less the tlen, and NOT less or equal !
fixes in the PR #270

Reply:
You are right @sebastienwarin 
This was referenced from a C code, o which a string[len] is a null char. Thanks for the check and benchmark on it.